plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -49,14 +49,15 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 func (f *FitError) Error() string {
 	var buf bytes.Buffer
 	buf.WriteString(fmt.Sprintf("pod (%s) failed to fit in any node\n", f.Pod.Name))
-	for node, predicates := range f.FailedPredicates {
-		reasons := make([]string, 0)
+	reasons := maps.NewCounter()
+	for _, predicates := range f.FailedPredicates {
 		for _, pred := range predicates {
-			reasons = append(reasons, pred.GetReason())
+			reasons.Incr(pred.GetReason())
 		}
-		reasonMsg := fmt.Sprintf("fit failure on node (%s): %s\n", node, strings.Join(reasons, ", "))
-		buf.WriteString(reasonMsg)
 	}
+	reasonMsg := fmt.Sprintf("fit failure summary on nodes : %v", reasons)
```

Comment:
Ok. I can make this change, but more for determinism/aggregation than dedupe, bc each event even if identical txt is unique based on the node being attempted ....